client/lib/jetpack/use-jp-presales-availability-query.ts

```diff
@@ -0,0 +1,68 @@
+import { useQuery } from '@tanstack/react-query';
```

Comment:
The vast majority of this file is being directly extracted from the jetpack presales component. This is just to make it more reusable